src/components/lib/cos_kernel_api.c

```diff
@@ -149,6 +158,62 @@ __untyped_bump_alloc(struct cos_compinfo *ci)
 	return __mem_bump_alloc(ci, 1, 0);
 }
 
+static int
+__cos_mem_alias_at(struct cos_compinfo *dstci, vaddr_t dst, struct cos_compinfo *srcci, vaddr_t src, int superpage)
+{
+	int order;
+	assert(srcci && dstci);
+
+	order = (superpage) ? SUPER_PAGE_ORDER : PAGE_ORDER;
+
+	if (call_cap_op(srcci->pgtbl_cap, CAPTBL_OP_CPY, src, dstci->pgtbl_cap, dst, order)) BUG();
+
+	return 0;
+}
+
+static vaddr_t
+__superpage_mem_bump_alloc(struct cos_compinfo *ci, vaddr_t addr, int order)
```

Comment:
this feels like it should be a generic function for each page size, but I could be wrong.